alpha-commons/src/main/java/at/ac/tuwien/kr/alpha/commons/programs/atoms/AtomQueryImpl.java

```diff
@@ -7,45 +7,49 @@
 import java.util.stream.Collectors;
 
 import at.ac.tuwien.kr.alpha.api.AnswerSet;
-import at.ac.tuwien.kr.alpha.api.AnswerSetQuery;
 import at.ac.tuwien.kr.alpha.api.programs.Predicate;
 import at.ac.tuwien.kr.alpha.api.programs.atoms.Atom;
-import at.ac.tuwien.kr.alpha.api.terms.ConstantTerm;
-import at.ac.tuwien.kr.alpha.api.terms.FunctionTerm;
-import at.ac.tuwien.kr.alpha.api.terms.Term;
+import at.ac.tuwien.kr.alpha.api.programs.atoms.AtomQuery;
+import at.ac.tuwien.kr.alpha.api.programs.terms.ConstantTerm;
+import at.ac.tuwien.kr.alpha.api.programs.terms.FunctionTerm;
+import at.ac.tuwien.kr.alpha.api.programs.terms.Term;
+import at.ac.tuwien.kr.alpha.commons.Predicates;
 
 /**
  * A query for ASP atoms matching a set of filter predicates.
  */
-public final class AnswerSetQueryImpl implements AnswerSetQuery {
+final class AtomQueryImpl implements AtomQuery {
 
 	private final Predicate predicate;
 	private Map<Integer, java.util.function.Predicate<Term>> filters = new HashMap<>();
 
-	private AnswerSetQueryImpl(Predicate pred) {
+	private AtomQueryImpl(Predicate pred) {
 		this.predicate = pred;
 	}
 
 	/**
-	 * Creates a new AnswerSetQuery that will match all atoms that are instances of the given {@link Predicate}.
+	 * Creates a new AtomQuery that will match all atoms that are instances of the given {@link Predicate}.
 	 * 
 	 * @param predicate the predicate to match against
-	 * @return a new AnswerSetQuery matching against the given predicate
+	 * @return a new AtomQuery matching against the given predicate
 	 */
-	public static AnswerSetQueryImpl forPredicate(Predicate predicate) {
-		return new AnswerSetQueryImpl(predicate);
+	static AtomQuery forPredicate(Predicate predicate) {
+		return new AtomQueryImpl(predicate);
+	}
+
+	static AtomQuery forPredicate(String name, int arity) {
+		return new AtomQueryImpl(Predicates.getPredicate(name, arity));
```

Comment:
Minor nitpicking here: why not chain the `forPredicate` methods, i.e., have here:
```suggestion
		return forPredicate(Predicates.getPredicate(name, arity));
```